### PR TITLE
Moves template_vars and template_paths to the policy objects

### PR DESCRIPTION
**Pull request by lorengordon**

### CODE REVIEW

**Comment by lorengordon** [comment]
Results of `make test`:
```
PASS
ok      terraform-aws-tardigrade-iam-principals/tests   443.993s
```

File: modules/roles/README.md [hunk 1]
```diff
@@ -6,9 +6,7 @@ Terraform module to create IAM roles
 <!-- BEGIN TFDOCS -->
 ## Requirements
 
-| Name | Version |
-|------|---------|
-| terraform | >= 0.12 |
+No requirements.
```
Comment (confusdcodr) [inline modules/roles/README.md:9]: Why was this removed?

File: modules/users/README.md [hunk 2]
```diff
@@ -6,9 +6,7 @@ Terraform module to create IAM users
 <!-- BEGIN TFDOCS -->
 ## Requirements
 
-| Name | Version |
-|------|---------|
-| terraform | >= 0.12 |
+No requirements.
```
Comment (confusdcodr) [inline modules/users/README.md:9]: Why was this removed?

File: modules/roles/README.md (hunk 1 above)
Comment (lorengordon) [reply modules/roles/README.md:9]: Just an automatic tfdocs thing, because the terraform version block was deleted. Seemed like just cruft and clutter, at this point.
